### DISCUSSION TITLE
How to add dynamic members in antmedia Android Conference

### DISCUSSION THREAD

**Discussion by pp1137** [ant-media/Ant-Media-Server#6523] [Q&A]
### Short description
Right now antmedia example provide only 4 member in conference i want to add multiple members in conference so required grid view for that so how we can do that in android sdk.

### Environment

Webrtc antmedia android sdk used

Used webrtc android sdk example.

### Steps to reproduce

Right now members video view added like below 

   .addRemoteVideoRenderer(player1Renderer, player2Renderer, player3Renderer, player4Renderer)
   
   but how to add it dynamically in gridview 

        webRTCClient = IWebRTCClient.builder()
                 .addRemoteVideoRenderer(player1Renderer, player2Renderer, player3Renderer, player4Renderer)
       .setLocalVideoRenderer(publisherRenderer)
                .setServerUrl(serverUrl)
                .setActivity(this)
                .setDataChannelEnabled(true)
                .setWebRTCListener(defaultConferenceListener)
                .setDataChannelObserver(createDatachannelObserver())
                .build();

### Expected behavior
- Required Multiple member show in gridview list.
- If we have 10 members in conference so how we can do it .

### Actual behavior
- Right now provided only 4 members conference.

 

### Logs
__Place logs on [pastebin](http://pastebin.com/) or elsewhere and put links here__

Ask your questions on [Ant Media Github Discussions](https://github.com/orgs/ant-media/discussions)

**Comment by burak-58** [comment]
Hi @pp1137,
You should override [this method ](https://github.com/ant-media/WebRTC-Android-SDK/blob/cdadbdc3a21cfc2af52a6a6b6054243dc33ba971/webrtc-android-framework/src/main/java/io/antmedia/webrtcandroidframework/api/DefaultWebRTCListener.java#L133) in DefaultWebRTCListener.java while providing it to WebRTCBuilder. In the overriden method provide the Video Views that you create in your application level. 

Let me convert this a github discussion. If we need any implementation we can convert is to issue back.

**Reply by pp1137** [reply]
Yes i need implementation sample code if you have please provide it.

**Reply by burak-58** [reply]
Hi @pp1137,
I meant any bug fix or new feature implementation in the SDK side after we continue on this discussion. I didn't mention about any custom (sample application) development. If you need such a custom development, please contact with contact@antmedia.io.

I suggest you that you may start by yourself as I told in my first response. It may be faster. If you have any issue with it you can ask here.

**Reply by pp1137** [reply]
hey @burak-58 

I need random list from member same like teams when 1 member added in group then shows full screen renderer, when 2 members in conference then shows screen size devide by 2 render view same like teams and other conference applications.

Do you have any ready code for that in android sdk. Please let me know if any guidance for that.

**Reply by burak-58** [reply]
Hi @pp1137,
In sample applications, our main focus is showing SDK usage and its APIs. So we don't have such a UI related codes that I can share with you. 
Besides, the renderer in our SDKs are based on View. So I believe that you can find resources to change a View dimensions dynamically.

**Comment by lastpeony** [comment]
Take a look at to the MultitrackPlay sample in android SDK
it demonstrates dynamic track adding to screen. You need to do similar implementation. 

I also created an issue to add a dynamic conference sample.

https://github.com/ant-media/Ant-Media-Server/issues/6625
